pkg/instrumentors/bpf/database/sql/bpf/probe.bpf.c

```diff
@@ -0,0 +1,90 @@
+#include "arguments.h"
+#include "span_context.h"
+#include "go_context.h"
+#include "go_types.h"
+
+char __license[] SEC("license") = "Dual MIT/GPL";
+
+#define MAX_QUERY_SIZE 100
+#define MAX_CONCURRENT 50
+
+struct sql_request_t {
+    u64 start_time;
+    u64 end_time;
+    char query[MAX_QUERY_SIZE];
+    struct span_context sc;
+    struct span_context psc;
+};
+
+struct {
+	__uint(type, BPF_MAP_TYPE_HASH);
+	__type(key, void*);
+	__type(value, struct sql_request_t);
+	__uint(max_entries, MAX_CONCURRENT);
+} context_to_sql_events SEC(".maps");
+
+struct {
+	__uint(type, BPF_MAP_TYPE_PERF_EVENT_ARRAY);
+} events SEC(".maps");
+
+// Injected in init
+volatile const bool should_include_db_statement;
+
+// This instrumentation attaches uprobe to the following function:
+// func (db *DB) queryDC(ctx, txctx context.Context, dc *driverConn, releaseConn func(error), query string, args []any)
+SEC("uprobe/queryDC")
+int uprobe_queryDC(struct pt_regs *ctx) {
+    // argument positions
+    u64 context_ptr_pos = 3;
+    u64 query_str_ptr_pos = 8;
+    u64 query_str_len_pos = 9;
+
+    struct sql_request_t sql_request = {0};
+    sql_request.start_time = bpf_ktime_get_ns();
+
+    if (should_include_db_statement) {
+        // Read Query string
+        void *query_str_ptr = get_argument(ctx, query_str_ptr_pos);
+        u64 query_str_len = (u64)get_argument(ctx, query_str_len_pos);
+        u64 query_size = MAX_QUERY_SIZE < query_str_len ? MAX_QUERY_SIZE : query_str_len;
+        bpf_probe_read(sql_request.query, query_size, query_str_ptr);
+    }
+
+    // Get goroutine as the key fro the SQL request context
```

Comment:
```suggestion
    // Get goroutine as the key for the SQL request context
```

Reply:
Comment removed after using the updated context propagation